nimiSlides.nim

```diff
@@ -176,57 +182,71 @@ template initReveal*() =
       body
 
-  template animateCode(lines: varargs[HSlice], body: untyped) =
+  template animateCode(lines: varargs[HSlice[int, int]], body: untyped) =
     ## Shows code and its output just like nbCode, but highlights different lines of the code in the order specified in `lines`.
     ## lines: Specify which lines to highlight and in which order. (Must be specified as a HSlice)
     ## Ex: 
     ## ```nim
     ## animateCode(1..1, 2..3, 5..5, 4..4): body
     ## ```
     ## This will first highlight line 1, then lines 2 and 3, then line 5 and last line 4.
-    discard
+    newNbBlock("animateCode", nb, nb.blk, body):
+      var linesString: string
+      if lines.len > 0:
+        linesString &= "|"
+      for line in lines:
+        linesString &= $line.a & "-" & $line.b & "|"
+      if lines.len > 0:
+        linesString = linesString[0 .. ^2]
+      nb.blk.context["highlightLines"] = linesString
+      captureStdout(nb.blk.output):
+        body
+
 
   template bigText(text: string) =
-    nbText: "<h2 class=\"r-fit-text\">" & text & "</h2>"
+    rawBlock: "<h2 class=\"r-fit-text\">" & text & "</h2>"
 
   template removeCodeOutput =
     if nb.blocks.len > 0:
       var blk = nb.blocks[^1]
-      if blk.kind == nbkCode:
+      if blk.command in ["nbCode", "animateCode"]:
         blk.output = ""
+        blk.context["output"] = ""
 
   template setSlidesTheme(theme: SlidesTheme) =
     nb.context["slidesTheme"] = ($theme).toLower
 
-  proc renderSlide(doc: NbDoc, slide: Slide): string =
-    let upper = 
-      if slide.pos.finish != -1: slide.pos.finish
-      else: doc.blocks.len - 1
-    
-    result = "<section>\n"
-    for i in slide.pos.start .. upper:
-      result &= doc.blocks[i].renderHtmlBlock(doc)
-    result &= "</section>\n"
-
-  proc renderReveal*(doc: NbDoc): string =
-    var content: string
-    for horiz in slidesCtx.sections:
-      content &= "<section>\n" # this is the top level section
-      for vertical in horiz:
-        # vertical corresponds to a single slide with many blocks. Must loop over them all and call `renderHTMLBlock` 
-        # if vertical.finish == -1: it is the last slide, grab the rest of all blocks
-        content &= doc.renderSlide(vertical)
-      content &= "</section>\n"
-    doc.context["slides"] = content
-    # This is neccecary because it will show the <span> tag otherwise:
-    result = "{{> document}}".render(doc.context).replace("<code class=\"nim hljs\">", "<code class=\"nim hljs\" data-noescape>")
-    result = result.replace("<pre><samp", "<pre style=\"width: 100%;\"><samp class=\"hljs\"") # add some background to code output block
-    result = result.replace("<pre>", "<pre style=\"width: 100%\">") # this makes code blocks a little bit wider
-
-  nb.render = renderReveal    
+  template useLocalReveal(path: string) =
+    discard
+    # set nb.partials["revealCSS/JS"]
+    # Should we set it relative to homeDir or srcDir?
 
 proc revealTheme*(doc: var NbDoc) =
```

Comment:
to simplify the reading flow you could move this to the top (before nbInitReveal, after all document, head, main, ...)